docs/branch_tracking.md

```diff
@@ -0,0 +1,53 @@
+# Branch tracking
+
+This requires basic knowledge on how Jujutsu's branches work[^1].
+
+As of December 2023 Jujutsu tracks branches by default, which is confusing 
+users coming from Git and Mercurial. To smoothen the transition branch tracking
+was introduced. 
+
+## Tracking a branch
+
+To track a branch permanently use `jj branch track <name-of-branch>`. It will 
+now be available in your local repository until you untrack it or it is 
+deleted on the remote. 
+
+Example:
+
+```sh
+$ # List all available branches, as we want our colleague's branch.
+$ jj git branch list 
```

Comment:
`jj branch list --all` to show all remote branches

Reply:
done.